uat/testing-features/src/main/resources/greengrass/features/ggmq-1.feature

```diff
@@ -0,0 +1,32 @@
+@GGAD
+Feature: GGMQ-1
+
+  As a developer, I can configure my client agents to connect and use MQTT
+
+  Background:
+    Given my device is registered as a Thing
+    And my device is running Greengrass
+    And my MQTT Client Control is running
+
+  Scenario: GGMQ-1-T1: As a customer, I can connect, subscribe and publish using client application to MQTT topic
+    When I create a Greengrass deployment with components
+      | aws.greengrass.clientdevices.Auth        | LATEST                                                        |
+      | aws.greengrass.clientdevices.mqtt.EMQX   | LATEST                                                        |
+      | aws.greengrass.clientdevices.IPDetector  | LATEST                                                        |
+      | aws.greengrass.client.Mqtt5JavaSdkClient | classpath:/greengrass/components/recipes/client_java_sdk.yaml |
+    Then the Greengrass deployment is COMPLETED on the device after 300 seconds
+    And agent "agent1" is connected to MQTT Client Control
+    When I create client device "test" with the following policy
+      | operation | resource     |
+      | connect   | "*"          |
+      | subscribe | "iot_data_0" |
+      | publish   | "iot_data_0" |
+    And I associate "test" with ggc
+    And I connect device "test" on "agent1" to "default_broker" as "agent1-test-default_broker"
+    Then connection "agent1-test-default_broker" is successfully established within 3 seconds
+    When I subscribe "agent1-test-default_broker" to "iot_data_0" with qos 0
+    Then subscription to "iot_data_0" is successfull on "agent1-test-default_broker"
+    When I publish "agent1-test-default_broker" to "iot_data_0" with qos 0 and message "Test message"
+    Then publish message "test" to "iot_data_0" is successfully on "agent1-test-default_broker"
```

Comment:
Let's simplify this: `Then the message is published with reason code 0`, for example.

Reply:
It is possible by using default logical connection name, but currently we want to explain architecture and possibilities in details.

With that architecture we can start several agents (identified by agentId string) and can create several MQTT connection on each.

in "agent1-test-default_broker" means logical connection id which can by just "connection1" for example or "last_connection" by default.

GADD examples have have just one MQTT connection. With that we can't test new MQTT5 features like group subscription.